### PR TITLE
chore: use Uri class for URI representation instead of String

### PR DESCRIPTION
**Pull request by ishche**
## How Has This Been Tested?

Analysis in progress notification should not contain uri encoding characters. (#2441)

## Checklist:
- [x] Each of my commits contains one meaningful change
- [x] I have performed rebase of my branch on top of the development
- [x] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] I have checked my code and corrected any misspellings

### CODE REVIEW

File: server/common/src/main/java/org/eclipse/lsp/cobol/common/model/Uri.java [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+import java.util.Objects;
+
+/** Uri container to enforce type checks*/
+@Slf4j
+public class Uri {
+    public static final Uri EMPTY = new Uri("");
+    // URI should be always decoded.
+    private final String uri;
+
+    public Uri(String uri) {
```
Comment (slavek-kucera) [inline server/common/src/main/java/org/eclipse/lsp/cobol/common/model/Uri.java:34]: How many of these will be created during a single processing?

Also https://www.slf4j.org/apidocs/src-html/org/slf4j/LoggerFactory.html#line-445

Comment (ishche) [reply server/common/src/main/java/org/eclipse/lsp/cobol/common/model/Uri.java:34]: Yep. logger is not needed there.
Do you have better suggestion than a wrapper class?

Comment (slavek-kucera) [reply server/common/src/main/java/org/eclipse/lsp/cobol/common/model/Uri.java:34]: Just avoid creating up to one per character in the worse case...

**Comment by ishche** [comment]
in favore of #2545

**Pull request closed without merging**